src/nvim/strings.c

```diff
@@ -413,21 +413,18 @@ char_u *vim_strchr(const char_u *string, int c)
 }
 
 /*
- * Version of strchr() that only works for bytes and handles unsigned char
- * strings with characters above 128 correctly. It also doesn't return a
- * pointer to the NUL at the end of the string.
+ * Search for c in buf. Appropriately handles values of c in [0, 255].
+ * This function preserves backward compatibility with previous vim code.
```

Comment:
Unless there is something special about this case (e.g. is this function at particular risk of being erroneously removed, compared to the hundreds of other such functions?), there is no need to mention compatibility.